### ISSUE TITLE
Adding sort functionality in Project and APK fragment.

### ISSUE THREAD

**Issue by chiragw15** [BuildmLearn/BuildmLearn-Toolkit-Android#212]
We can add sort functionality in Project and APK fragment. Sorting can done be based on Name of Project/APK , modified Date and Author.

**Comment by croozeus** [comment]
Instead of sorting, allowing to group the projects under a title / subject would be a good enhancement. Teachers or parents who are creating apps for students / kids may want to group the apps by subject or different topics.

**Comment by codingblazer** [comment]
Hey @croozeus ,Can I work on the above mentioned enhancement ?

**Comment by croozeus** [comment]
On the one I proposed, yes!

**Comment by codingblazer** [comment]
Thanks @croozeus. Will be sending a PR soon.

**Comment by chiragw15** [comment]
@croozeus Sure, I'll edit the issue. Also, @codingblazer I have started the work on this issue already, so please can you work on something else. Thanks

**Comment by codingblazer** [comment]
Hey @chiragw15 , I am been assigned on the new issue reported by @croozeus . You can definitely carry on with your work on original issue.Thanks.

**Comment by chiragw15** [comment]
Sorry but he asked to edit this issue only to change it from sorting to grouping, which I did.

**Comment by codingblazer** [comment]
Hey ,@chiragw15 .I am confused here. So, should I work on this as assigned ?

**Comment by chiragw15** [comment]
@codingblazer Sorry but please can you work on some other issue. I have edited the issue as suggested by croozeus and I was already working on this issue.

**Comment by codingblazer** [comment]
@chiragw15 You can proceed. However, the new issue is completely different from the original one.

**Comment by chiragw15** [comment]
@codingblazer Sorry but I already started the work . If I wouldn't have , then I would have told you to proceed at once. Sorry for trouble.

**Comment by croozeus** [comment]
@chiragw15 You can work on the updated issue. Thanks for the edit. Once you start work on an issue, please assign it to yourself - this way everyone knows that you are working on this. Also, would be good to comment once you start working on an issue and update progress.

@codingblazer Since chirag has already started work on this. You can pick another issue :-)

**Comment by croozeus** [comment]
@chiragw15 btw, it would be great if you could first show us a design of how you'll implement the groups before actually implementing it. We all can review the flow first.

**Comment by chiragw15** [comment]
Thanks @croozeus but I don't think we can assign issues to ourselves. It is project maintainers or owners who can assign the issues to people. I will comment on it next time I create an issue.

**Comment by chiragw15** [comment]
Yes sure , I'll do it definitely :+1:

**Comment by chiragw15** [comment]
@croozeus @opticod How about something like this to group projects under a title?

![screenshot_20161113-042658](https://cloud.githubusercontent.com/assets/16978820/20242607/82aeb3de-a95a-11e6-9a1c-674c95af4d1a.png)

**Comment by croozeus** [comment]
Thanks for the mockup @chiragw15 

Couple of questions:

1. How would the user create a group initially?
2. Where would the user be asked which group to save the app to?
3. How can the user move the app to another group? 
4. How would the user rename a group, what would the interaction and interface for this look like?

If you could show the user flow for these scenarios, it would be great. We should have all of these figured out before you start implementing.

**Comment by chiragw15** [comment]
@croozeus  The flow will be exactly like a normal directory (file manager) system. Just assume group to be a folder and project to be a file. 

Ans 1.) There will be an option to create a new group in the Project Fragment. 
Ans 2.) While saving a project, user will be asked where he wants to save the project. There will also be an option to create a new group and and save a project in it. ( Just like normal saving option in a computer).
Ans 3.) While long pressing a project , options will be given to copy the project, move the project, delete the project. (Just like, what right click button does).
Ans 4.) While long pressing on a group, option will be there to rename it, delete it etc.

I am working on the flow only, and will show you as soon I complete it.

**Comment by chiragw15** [comment]
@croozeus This is the appflow I am thinking of. Please have a look. Waiting for feedback :)

https://drive.google.com/open?id=0B6IO4qTKhVUCdWxNTk1pUlFVd0E

**Comment by croozeus** [comment]
@chiragw15  Thanks for the mocks. The overall flow looks good to me.

For steps 3, 4 and 5 in the app flow - what I don't like is the overlapping dialogs i.e. "Create Group" modal view over "Select Location" modal view. I would not make "Select Location" a dialog / modal view - rather let it take the complete screen as "Info template" screen.

Also, there should be some preconfigured groups that we should add to solve the empty room problem. 

Please update when you start coding, so that if you are busy someone else can take this up.

**Comment by chiragw15** [comment]
@croozeus Thanks for the feedback :) . I'll change "Select Location" dialog box to full screen .
>Also, there should be some preconfigured groups that we should add to solve the empty room problem.

For this, how about groups with title as the type of app template , like InfoApps, QuizApps, VideoApps, ComprehensionApps, DictationApps etc.
I was just waiting for your feedback. I'll start coding now :).

**Comment by croozeus** [comment]
> For this, how about groups with title as the type of app template , like InfoApps, QuizApps, VideoApps, ComprehensionApps, DictationApps etc.
> I was just waiting for your feedback. I'll start coding now :).

I don't think we should group them by template type. It wouldn't make sense to ask the user which group he wants to add to if we do template wise grouping. It should be on a title / topic based type for example, some teachers may want to group the app by chapters or subjects.

**Comment by codingblazer** [comment]
@chiragw15 Are you working on this ?

**Comment by chiragw15** [comment]
@codingblazer I have completed 70-80% of the work. I was out of town for last 10 days, just came back yesterday. I will resume my work from today.

**Comment by opticod** [comment]
@chiragw15, Is there any updates regarding this?

**Comment by chiragw15** [comment]
@opticod Yup, I tried this with two different approaches and finally succeeded with one. I have done almost everything. Just a little bit of tweaks are left. Will test the app on various devices to check everything is working fine or not. Will send a PR after making sure everything is working fine and the app is not crashing.
